### DISCUSSION TITLE
It possible to use Teams-AI with C# instead of JS?

### DISCUSSION THREAD

**Discussion by YuriyMorozyuk95** [microsoft/teams-sdk#165] [Q&A] [answered]
It possible to use Teams-AI with C# instead of JS, or just use as API, or what alternatives you recommend (CLU, Semantic-kernel, ect)?

**Comment by YuriyMorozyuk95** [comment]
As I understood Teams-AI, using OpenAI or Azure OpenAI behind, could you recommend some C# framworks to work with it?

**Reply by YuriyMorozyuk95** [reply]
After checking repository I found that Dotnet version underdevelopment could you please tell approximate time when it will be released?

**Reply by palonza** [reply]
Hi @YuriyMorozyuk95,

Answering your initial question: it would be good if you were more specific about your requirements, anyway in .net you have the following AI frameworks:

ML.NET
Accord.NET
TensorFlow.NET

**Reply by YuriyMorozyuk95** [reply]
Thank you for answer, does Ml.Net, Accord.Net, TensorFlow.Net work with OpenAI or Azure OpenAI ?

From what I cheeck in this project I see that there is couple branches with dotnet impementation of Teams-AI, do there is approximate datetime when dotnet implementation will be ready? As I am interesting to use it in teams hackaton.

**Reply by YuriyMorozyuk95** [reply]
I am interesting in using some tool to build teams bot on C# with Conversational Language Understanding.

**Comment by corinagum** [comment, marked as answer]
Hi @YuriyMorozyuk95, dotnet is under active development and we hope to get preview out in the next few months! :)

**Reply by corinagum** [reply]
@YuriyMorozyuk95  dotnet is now in private preview! go check it out and let us know your feedback@
